modules/space/rendering/renderablestars.cpp

```diff
@@ -654,132 +574,99 @@ RenderableStars::RenderableStars(const ghoul::Dictionary& dictionary)
+    addProperty(_filterOutOfRange);
 
 
-    if (p.sizeComposition.has_value()) {
-        _psfMultiplyOption =
-            static_cast<int>(codegen::map<SizeComposition>(*p.sizeComposition));
-    }
-    else {
-        _psfMultiplyOption = 5;
+    auto markTextureAsDirty = [this]() {_pointSpreadFunctionTextureIsDirty = true; };
+    _halo.texturePath = absPath(p.halo.texture.string()).string();
+    _halo.texturePath.onChange(markTextureAsDirty);
+    _halo.file = std::make_unique<ghoul::filesystem::File>(_halo.texturePath.value());
+    _halo.file->setCallback(markTextureAsDirty);
+    _halo.container.addProperty(_halo.texturePath);
+    _halo.opacity = p.halo.partialOpacity.value_or(_halo.opacity);
+    _halo.container.addProperty(_halo.opacity);
+    _halo.scale = p.halo.scale.value_or(_halo.scale);
+    _halo.container.addProperty(_halo.scale);
+    addPropertySubOwner(_halo.container);
+
+    if (p.glare.has_value()) {
+        _glare.texturePath = absPath(p.glare->texture.string()).string();
+        _glare.file =
+            std::make_unique<ghoul::filesystem::File>(_glare.texturePath.value());
+        _glare.file->setCallback(markTextureAsDirty);
+        _glare.opacity = p.glare->partialOpacity.value_or(_glare.opacity);
+        _glare.scale = p.glare->scale.value_or(_glare.scale);
     }
+    _glare.texturePath.onChange(markTextureAsDirty);
+    _glare.container.addProperty(_glare.texturePath);
+    _glare.container.addProperty(_glare.opacity);
+    _glare.container.addProperty(_glare.scale);
+    addPropertySubOwner(_glare.container);
 
-    _parametersOwner.addProperty(_psfMultiplyOption);
-    _parametersOwner.addProperty(_lumCent);
-    _parametersOwner.addProperty(_radiusCent);
-    _parametersOwner.addProperty(_brightnessCent);
 
     _magnitudeExponent = p.magnitudeExponent.value_or(_magnitudeExponent);
-    _parametersOwner.addProperty(_magnitudeExponent);
-
-    auto renderPsf = [this]() { renderPSFToTexture(); };
-
-    _spencerPSFParamOwner.addProperty(_p0Param);
-    _p0Param.onChange(renderPsf);
-    _spencerPSFParamOwner.addProperty(_p1Param);
-    _p1Param.onChange(renderPsf);
-    _spencerPSFParamOwner.addProperty(_p2Param);
-    _p2Param.onChange(renderPsf);
-    _spencerPSFParamOwner.addProperty(_spencerAlphaConst);
-    _spencerAlphaConst.onChange(renderPsf);
+    addProperty(_magnitudeExponent);
 
-    _moffatPSFParamOwner.addProperty(_FWHMConst);
-    _FWHMConst.onChange(renderPsf);
-    _moffatPSFParamOwner.addProperty(_moffatBetaConst);
-    _moffatBetaConst.onChange(renderPsf);
 
-    _parametersOwner.addPropertySubOwner(_spencerPSFParamOwner);
-    _parametersOwner.addPropertySubOwner(_moffatPSFParamOwner);
+    _parameters.sizeComposition.addOptions({
+        { AppBrightness, "Use Star's Apparent Brightness" },
+        { LumSize, "Use Star's Luminosity and Size" },
+        { LumSizeAppBrightness, "Luminosity, Size, App Brightness" },
```

Comment:
```suggestion
        { LumSizeAppBrightness, "Luminosity, Size, Apparent Brightness" },
```

If we want to be consistent with using the full word as in the other strings.  However, maybe it becomes too long?

Reply:
This composition method is no more